projects/ion/src/lib/step/step.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { Component } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { render, screen } from '@testing-library/angular';
+import { IonIconComponent } from '../icon/icon.component';
+import { StepComponent, StepType } from './step.component';
+
+const defaultValue: StepType[] = [
+  {
+    label: 'Step 1',
+  },
+  {
+    label: 'Step 2',
+  },
+  {
+    label: 'Step 3',
+  },
+];
+
+describe('Static StepComponent', () => {
+  const sut = async (
+    customProps: StepType[] = defaultValue
+  ): Promise<HTMLElement> => {
+    await render(StepComponent, {
+      componentProperties: {
+        steps: customProps,
+      },
+      declarations: [IonIconComponent],
+      imports: [FormsModule],
+    });
+    return screen.findByTestId('ion-step');
+  };
+
+  it('should render step component with 3 steps', async () => {
+    await sut();
+    expect(screen.getByText('Step 3')).toBeTruthy();
```

Comment:
This spec does not guarantee the 3 steps rendered, you can use it.each with defaultValue to interable